file_io.cpp

```diff
@@ -43,7 +69,7 @@ void FileClose(fileTYPE *file)
 			file->type = 0;
 		}
 	}
-	file->fd = -1;
+	file->filp = nullptr;
 }
 
 int FileOpenEx(fileTYPE *file, const char *name, int mode, char mute)

```

Comment:
I can create two helper functions for `FileOpenEx`, one for zipped files and the other for non-zipped and then simply call them from here if you think that is cleaner?